airflow/providers/slack/hooks/slack.py

```diff
@@ -113,42 +111,31 @@ class SlackHook(BaseHook):
 
     def __init__(
         self,
-        token: str | None = None,
-        slack_conn_id: str | None = None,
+        *,
+        slack_conn_id: str,
```

Comment:
`default_conn_name` is not really used here.

Reply:
I guess it is never use any default connection ID
https://airflow.apache.org/docs/apache-airflow-providers-slack/stable/connections/slack.html#default-connection-ids

Reply:
I proposed a fix for it in this PR: https://github.com/apache/airflow/pull/34548

Reply:
Better say feature. Without removal in this PR it was quite difficult to do that.
Let's move discussion in to your PR

Reply:
Yes, it's more an improvement of course. :)